tests/unit/geo.tcl

```diff
@@ -374,53 +402,84 @@ start_server {tags {"geo"}} {
             r del mypoints
 
             if {[randomInt 10] == 0} {
+                if {$type == "byradius"} {
+                    set radius_km [expr {[randomInt 5000]+10}]
+                } elseif {$type == "bybox"} {
+                    set width_km [expr {[randomInt 5000]+10}]
+                    set height_km [expr {[randomInt 5000]+10}]
+                }
                 # From time to time use very big radiuses
-                set radius_km [expr {[randomInt 50000]+10}]
             } else {
                 # Normally use a few - ~200km radiuses to stress
                 # test the code the most in edge cases.
-                set radius_km [expr {[randomInt 200]+10}]
+                if {$type == "byradius"} {
+                    set radius_km [expr {[randomInt 200]+10}]
+                } elseif {$type == "bybox"} {
+                    set width_km [expr {[randomInt 200]+10}]
+                    set height_km [expr {[randomInt 200]+10}]
+                }
+            }
+            if {$rv ne {}} {
+                set radius_km [lindex $rv 1]
+                set width_km [lindex $rv 1]
+                set height_km [lindex $rv 1]
```

Comment:
i have a feeling that it would be cleaner to just have these converted to meters, and don't mention `km` anywhere in the code (except for the comment above).